### DISCUSSION TITLE
Allow disabling client-side validation on some fields

### DISCUSSION THREAD

**Discussion by BenoitAverty** [directus/directus#17616] [Feature Requests]
### Summary

The feature would be to have a checkbox in the field configuration that completely disables the client-side validation of the field in the data studio.

If the checkbox is checked, then only server-side validation is performed (and the database would check too of course). Client-side validation is completely bypassed.

### Motivation

If I'm not mistaken, values entered in the form for creating/updating items are validated client-side and server-side according to the collection configuration.

My use-case is the following : one of the fields in a collection I'm managing is computed in a custom hook extension based on the values of other fields. In the DB, this field should be non-null because the extension will ensure there is always a value. However, I'd like to forbid the user to set a value in this field at creation time.

The problem is that I can't have it both ways : 
 - If I set the field to not allow null values, then I can't submit the form without filling this field (client-side)
 - If I set the field to allow null values, then my DB could be corrupted in the future and looking at the schema could confuse other developers.

I would like the user to be able to leave the field blank, but have a non-null column in the db. The hook will ensure there is always a value, and if not the server-side validation would prevent an invalid value anyway. 

### Detailed Design

I don't know how this could be implemenbted but it would probably be slightly complicated depending on the order of operations. If server-side validation is performed before custom filters then this would not allow me to do what I'm doing anyways.

### Requirements List

# Must have

A checkbox that disables client-side validation of a field

### Drawbacks

I don't really see one becase the feature is completely optional. 

Also, I'm not sure client-side and server-side validation is exactly the same. That would be a shame if some errors became less clear because the DB error bubbles back to the user. Or worse, if some validation is only performed client-side (but I don't think that's the case because there's always the rest api)

### Alternatives

Completely separate validation and column definition, making a non-null column mandatory by default but allowing to override this. But I think this would be worse because we d'ont want to see raw DB errors as the user of the data studio.

### Adoption Strategy

No breaking change, the default would be the current behaviour because the only use case I can see is in case of custom hook or filter flow.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
